src/spectrecoff/Output.fs

```diff
@@ -6,59 +6,61 @@ open Spectre.Console
 open System
 
 // Styles
-let mutable calmStyle = None
-let mutable calmColor = Color.SteelBlue
+let mutable calmLook: Look = 
+    { Color = Color.SteelBlue
+      Decoration = [ Decoration.None ] }
+let mutable pumpedLook: Look = 
+    { Color = Color.DeepSkyBlue3_1
+      Decoration = [ Decoration.Italic ]}
 
-let mutable pumpedStyle = Italic
-let mutable pumpedColor = Color.DeepSkyBlue3_1
+let mutable edgyLook: Look = 
+    { Color = Color.DarkTurquoise
+      Decoration = [ Decoration.Bold ] }
 
-let mutable edgyStyle = Bold
-let mutable edgyColor = Color.DarkTurquoise
+let mutable linkLook = 
+    { Color = pumpedLook.Color
+      Decoration = [ Decoration.Underline; Decoration.Italic ] }
 
-let mutable linkColor = pumpedColor
-
-// Special strings
 let mutable bulletItemPrefix = " + "
 
-// Basic output
-let private markupWithColor color content =
-    $"[{color}]{Markup.Escape content}[/]"
-
-let private markupWithStyle style content =
-    $"[{style}]{Markup.Escape content}[/]"
-
-let private markupWithColorAndStyle color (style: string) content =
-    $"[{color} {style}]{Markup.Escape content}[/]"
-
 let private padEmoji (emoji: string) =
     match emoji.StartsWith ":" with
     | true -> emoji
     | false -> $":{emoji}:"
 
-let markupString (color: Color option) (style: Layout.Style option) content =
-    match style with
-    | None ->
-        match color with
-        | None -> content
-        | Some c -> markupWithColor c content
-    | Some s ->
-        match color with
-        | None -> markupWithStyle s content
-        | Some c -> markupWithColorAndStyle c (stringifyStyle s) content
-
-let pumped content = markupString (Some pumpedColor) (Some pumpedStyle) content
-let edgy content = markupString (Some edgyColor) (Some edgyStyle) content
-let calm content = markupString (Some calmColor) calmStyle content
-
-let printMarkedUpInline content = AnsiConsole.Markup $"{content}"
-let printMarkedUp content = AnsiConsole.Markup $"{content}{Environment.NewLine}"
-
 let rec private joinSeparatedBy (separator: string) (strings: string list) =
     match strings with
     | [] -> ""
     | [s] -> s
     | head::tail -> head + separator + joinSeparatedBy separator tail
 
+let private stringify (decorations: Decoration list) = 
+    decorations |> List.map (fun s -> s.ToString())
+
+let private css colorOption decorations = 
```

Comment:
I am confused about this `css` method name. As far as I can see, this conditional concatenation has nothing to do with css, can you explain?

Reply:
I used soo many names and variation of `toMarkup*` that I got frustrated, hehe. But I'll find a better name.